hw/ip/spi_host/doc/_index.md

```diff
@@ -660,16 +659,613 @@ To use this reset, assert {{< regref "CONTROL.SW_RST" >}}, and then wait for the
 
 {{< incGenFromIpDesc "../data/spi_host.hjson" "hwcfg" >}}
 
-## Design Details
+# Design Details
+
+## Component Overview
+
+Transaction data words flow through the SPI_HOST IP in a cycle which starts with the TX FIFOs, shown in the block diagram above.
+At the output of the TX FIFOs each data word is separated into individual bytes by the Byte Select block, which is also responsible for parsing the byte-enable mask and discarding unwanted bytes.
+Selected bytes are then passed into the shift register, where they are played out at Standard, Dual, or Quad speed.
+For receive segments, outputs from the shift register are passed into the Byte Merge block to be packed into 32-bit words.
+Finally the repacked words are inserted into the RX FIFO to be read by firmware.
+
+All of the blocks in the data path use ready-valid handshakes for flow control.
+In addition, the Byte Select block expects a `flush` pulse from the shift register to signify when no further data is needed for the current segment, and so any remaining data in the current word can be discarded.
+Likewise, the Byte Merge block receives a `last` signal from the shift register to identify the end of a command segment so that any partial words can be passed into the RX FIFO (regardless of whether the last byte forms a complete 32-bit word).
+The shift register is then responsible for driving and receiving data on the `cio_sd` lines.
+It coordinates all of the data flow to and from the Byte Select and Byte Merge blocks.
+
+The SPI_HOST FSM parses the software command segments and orchestrates the proper transmission of data through its control of the shift register.
+The FSM directly drives the `cio_sck` and `cio_csb` signals at the commanded speed.
+It also controls the shift register: dictating the correct timing for sending out each beat of data, loading bytes from the Byte Select, and sending bytes on to the Byte Merge block.
+
+## RX and TX FIFOs
+
+The TX and RX FIFOs store the transmitted and received data, while also serving as the clock domain crossing for SPI data.
+So in each direction there is at least one asynchronous FIFO, which serves as the CDC.
+The RX FIFO is 32 bits wide, matching the width of the TLUL register bus.
+The TX FIFO on the other hand is 36 bits wide, with 32 bits of SPI data (again to match the TLUL bus width) plus 4 byte enable-bits, which are passed into the core to allow the processing of unaligned writes.
+
+The depth of these FIFOs is controlled by two independent parameters for the RX and TX queues.
+Since the depth of asynchronous FIFOs is usually limited to powers of two, the TX or RX queue may consist of *two* FIFO components: an mandatory asynchronous FIFO and a possible synchronous component, connected in series.
```

Comment:
```suggestion
Since the depth of asynchronous FIFOs is usually limited to powers of two, the RX or TX queue may consist of *two* FIFO components: an mandatory asynchronous FIFO and a possible synchronous component, connected in series.
```